### PR TITLE
[Enhancement] Improve Metis client test quality

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context

This PR addresses the client test speed/quality improvements as discussed in our developer meetings. [This table](https://confluence.ase.in.tum.de/display/ArTEMiS/Test+suite+speed+improvements) documents the current state on this topic.

### Description

- I removed importing production code and used Mocks where needed.
- I removed `schemas: [NO_ERROR_SCHEMA]` which implied that errors in the template were not checked.

### Steps for Testing

Review the test code.

### Review Progress

- Code Review
  - [x] Review 1
  - [x] Review 2

### CODE REVIEW

**Review by Olimbia23 — DISMISSED** [review]
Nice work @lschlesinger. 
Code changes look good. Approve 👍

File: src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts [hunk 1]
```diff
@@ -27,7 +29,7 @@ describe('PostComponent', () => {
     beforeEach(() => {
         return TestBed.configureTestingModule({
             imports: [],
-            declarations: [PostComponent, MockPipe(HtmlForMarkdownPipe)],
+            declarations: [PostComponent, MockPipe(HtmlForMarkdownPipe), MockComponent(PostHeaderComponent), MockComponent(PostFooterComponent)],
             schemas: [NO_ERRORS_SCHEMA],
```
Comment (MerlinMehmed) [inline src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts:33]: It looks like you might have forgotten to remove the NO_ERRORS_SCHEMA from this file, or is there another reason for not doing it?

**Review by MerlinMehmed — COMMENTED** [review]
Changes look okay, I left one small comment

**Review by coolchock — APPROVED** [review]
code looks good to me :+1:

File: src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts (hunk 1 above)
Comment (lschlesinger) [reply src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts:33]: totally true, I forgot it. I removed it, tests still pass. Thanks for the hint!

**Review by MerlinMehmed — APPROVED** [review]
Changes still look good to me

File: src/test/javascript/spec/component/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-tag-selector.component.spec.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ describe('PostTagSelectorComponent', () => {
             imports: [],
             providers: [{ provide: MetisService, useClass: MockMetisService }],
             declarations: [PostTagSelectorComponent, MockPipe(ArtemisTranslatePipe)],
-            schemas: [NO_ERRORS_SCHEMA],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-tag-selector.component.spec.ts:29]: This is still using `CUSTOM_ELEMENTS_SCHEMA`?

File: src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts [hunk 3]
```diff
@@ -27,8 +29,7 @@ describe('PostComponent', () => {
     beforeEach(() => {
         return TestBed.configureTestingModule({
             imports: [],
-            declarations: [PostComponent, MockPipe(HtmlForMarkdownPipe)],
-            schemas: [NO_ERRORS_SCHEMA],
+            declarations: [PostComponent, MockPipe(HtmlForMarkdownPipe), MockComponent(PostHeaderComponent), MockComponent(PostFooterComponent)],
             providers: [],
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts:33]: Bonus points if you remove the lines with empty arrays `[]`, as they shouldn't be needed.

File: src/test/javascript/spec/component/shared/metis/postings-footer/post-footer/post-footer.component.spec.ts [hunk 4]
```diff
@@ -25,8 +28,7 @@ describe('PostFooterComponent', () => {
         return TestBed.configureTestingModule({
             imports: [],
             providers: [],
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/shared/metis/postings-footer/post-footer/post-footer.component.spec.ts:30]: Same here.

File: src/test/javascript/spec/component/shared/metis/postings-footer/answer-post-footer/answer-post-footer.component.spec.ts [hunk 5]
```diff
@@ -34,8 +36,7 @@ describe('AnswerPostFooterComponent', () => {
         return TestBed.configureTestingModule({
             imports: [],
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/shared/metis/postings-footer/answer-post-footer/answer-post-footer.component.spec.ts:37]: And here.

File: src/test/javascript/spec/component/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-tag-selector.component.spec.ts (hunk 2 above)
Comment (lschlesinger) [reply src/test/javascript/spec/component/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-tag-selector.component.spec.ts]: Nope, I've overlocked it. I changed it in a549ab0

**Comment by lschlesinger** [comment]
@FrankeLukas Thanks for checking the code changes... as I like bonus points, I removed the empty brackets everywhere, even found some more places in my test files :)

**Review by jpbernius — APPROVED** [review]

**Pull request merged**